File: src/test/java/2023/unsorted/easy/LengthOfLastWordTest.java
package easy;

import org.example.may.easy.LengthOfLastWord;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class LengthOfLastWordTest {
    private final LengthOfLastWord length = new LengthOfLastWord();

    @Test
    public void twoWords() {
        String s = "Hello World";
        Assertions.assertEquals(5, length.lengthOfLastWord(s));
    }

    @Test
    public void fiveWords() {
        String s = "   fly me   to   the moon  ";
        Assertions.assertEquals(4, length.lengthOfLastWord(s));
    }

    @Test
    public void fourWords() {
        String s = "luffy is still joyboy";
        Assertions.assertEquals(6, length.lengthOfLastWord(s));
    }
}

File: leetcode-python/src/test/easy/length_of_last_word_test.py
import pytest

from src.solution.easy.length_of_last_word import length_of_last_word


@pytest.mark.parametrize("exp, word", [
    [5, "Hello World"],
    [4, "   fly me   to   the moon  "],
    [6, "luffy is still joyboy"]
])
def test_length_of_last_word(exp: int, word: str) -> None:
    assert exp == length_of_last_word(word)


if __name__ == '__main__':
    pytest.main()
